### PR TITLE
14668 logo quexml export

### PR DESCRIPTION
**Pull request by mbischof**
Fixed issue # : 
New feature # :
Changed feature # :
Dev: 
Dev:

### CODE REVIEW

File: application/libraries/admin/quexmlpdf.php [hunk 1]
```diff
@@ -3806,6 +3806,16 @@ protected function newPage($init = false)
             $this->SetHeaderMargin(0);
             $this->SetFooterMargin(0);
 
+            $oTemplate = Template::model()->getInstance();
```
Comment (olleharstedt) [inline application/libraries/admin/quexmlpdf.php]: Could you please provide instructions on how to (manually) test this code? Something like

1. Go to x
2. Click on y
3. See z
4. ...

Thank you.

File: application/libraries/admin/quexmlpdf.php [hunk 2]
```diff
@@ -3806,6 +3806,16 @@ protected function newPage($init = false)
...
+            $sLogoFileName = $oTemplate->filesPath . Yii::app()->getConfig('pdflogofile');
+            if (file_exists($sLogoFileName)) {
+                $result = LSYii_ImageValidator::validateImage($sLogoFileName);
+                if (isset($result['check']) && $result['check'] == true) {
+                    $sLogo = "@" . file_get_contents($sLogoFileName);
```
Comment (olleharstedt) [inline application/libraries/admin/quexmlpdf.php]: You probably want to check if `file_get_contents` returned anything valid.

Comment (Shnoulle) [reply application/libraries/admin/quexmlpdf.php]: Done in `$result = LSYii_ImageValidator::validateImage($sLogoFileName);` :)

File: application/libraries/admin/quexmlpdf.php (hunk 1 above)
Comment (Shnoulle) [reply application/libraries/admin/quexmlpdf.php]: Think it's hard to test pdf file ? Check if logo is shown in pdf file ?

**Comment by Shnoulle** [comment]
Question : did logo must be same for queXML than from pdf ?

In my opinion : logo for pdf and queXML pdf can be different. Maybe adding a new global settings ?

File: application/libraries/admin/quexmlpdf.php (hunk 1 above)
Comment (mbischof) [reply application/libraries/admin/quexmlpdf.php]: In your upload/themes/survey/{...}/files - folder must be the image defined in your application/config/config-defaults.php `pdflogofile` setting, then pdf queXML export (responses/sa/browse, second icon in list view) should display the image in the pdfs header

Comment (Shnoulle) [reply application/libraries/admin/quexmlpdf.php]: Yes : but @olleharstedt talk for a way to add a [test](https://github.com/LimeSurvey/LimeSurvey/tree/develop/tests) for this. I don't think there are automatic way to add a test to check PDF content.

Comment (Shnoulle) [reply application/libraries/admin/quexmlpdf.php]: Arg … sorry … **to (manually) test ** sorry both … please : delete my comments ;)

File: application/libraries/admin/quexmlpdf.php (hunk 2 above)
Comment (olleharstedt) [reply application/libraries/admin/quexmlpdf.php]: No? It can return `false`, which should be checked.

**Comment by maziminke** [comment]
@c-schmitz , can we please merge this one? Its details were discussed long time ago (see https://bugs.limesurvey.org/view.php?id=14668).
Thanks!

**Comment by olleharstedt** [comment]
I'd suggest not to merge any new features without:

1) Manual QA
2) Accepting tests

Sorry for the bureaucracy, but we're trying to come up with a more stable workflow for new features, both internally and externally. Also quite busy making LS4 bug free, so I didn't have time yet to test this.

**Pull request merged**